src/modules/seasons/seasons.service.ts

```diff
@@ -1,16 +1,40 @@
+import { SeasonEntity } from 'src/entities/season.entity';
+import { CreateSeasonDto, SeasonDto } from './seasons.types';
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
-import { Season } from '../../entities/season.entity';
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
 
 @Injectable()
 export class SeasonsService {
     constructor(
-        @InjectRepository(Season)
-        private seasonsRepository: Repository<Season>
-    ) {}
+        @InjectRepository(SeasonEntity)
+        private seasonsRepository: Repository<SeasonEntity>,
+        @InjectMapper() private readonly classMapper: Mapper,
+    ) { }
 
-    findAll(): Promise<Season[]> {
-        return this.seasonsRepository.find();
+    async findAll(): Promise<SeasonDto[]> {
+        try {
+            return this.classMapper.mapArrayAsync(await this.seasonsRepository.find(), SeasonEntity, SeasonDto);
+        } catch (err) {
+            throw new Error(`findAll failed: ${err.message}`);
+        }
+    }
+
+    async findOne(id: number): Promise<SeasonDto> {
+        try {
+            return this.classMapper.mapAsync(await this.seasonsRepository.findOne({  where: { id }}), SeasonEntity, SeasonDto)
+        } catch (err) {
+            throw new Error(`findOne failed: ${err.message}`);
```

Comment:
er want to specfically throw an `InternallException` error here, since we don't want postgres errors to make it to the requestor. 

We should also look into a logging system in another PR so that we can...

1. log the error
2. return internal server error the FE